docs/how-to-chain-with-pipeline.md

```diff
@@ -0,0 +1,282 @@
+- ***Task-level provenance***: A SLSA provenance that Tekton Chains generates to only include the details of a particular TaskRun execution. It's particularly needed for a standalone TaskRun that is not spawned by a PipelineRun. By contrast, if it's a child TaskRun of a PipelineRun, Task-level provenance will miss the details of other TaskRuns within that Pipeline.
+- ***Input Artifacts***: A canonical term used in this doc to refer to the artifacts that influenced the build process such as source code repository, dependencies and so on. It's mapped to `resolvedDependences` field in SLSA v1.0, and mapped to `materials` field in SLSA v0.1 & v0.2.
+- ***Output Artifacs***: A canonical term used in this doc to refer to the artifacts that the build process produced i.e. an OCI image. This is mapped to `Subjects` field in all SLSA versions.
+- ***`Results`***: `Results` are Tekton API fields that authors can use to emit some information after a TaskRun/PipelineRun is complete. `Results` can be used to pass along information to different tasks within a pipeline or aggregate different task results to a pipeline result. Check out [Tekton official doc](https://tekton.dev/docs/pipelines/pipelines/#using-results) more information. *Note: API result field is completely different from [Tekton Results Operator](https://tekton.dev/docs/results/)*.
+- ***Type hinting***: Refer to specially named results/params that aim to enable Tekton Chains to understand the input artifacts and outputs of a PipelineRun/TaskRun.
+
+
+## How does Tekton Chains work?
+
+Tekton Chains works by reconciling the run of a task or a pipeline. Once the run is observed as `completed`, Tekton Chains will take a snapshot of the completed TaskRun/PipelineRun, and start its core works in the order of ***`formatting`*** (generate provenance json) -> ***`signing`*** (sign the payload using the key configured by user) -> ***`uploading`*** (upload the provenance and its signature to the storage configured by user).
+
+![](../images/how-chains-works.png)
+
+## How to configure Tekton Chains
+Tekton Chains supports both SLSA v0.2 and v1.0 provenance for both task-level and pipeline-level provenance. 
+
+The following shows the mapping between slsa version and formatter name.
+- SLSA v1.0 provenance: `slsa/v2alpha2`
+- SLSA v0.2 provenance: `slsa/v1` or `in-toto`
+
+To select Task-level provenance version
+
+```shell
+kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"artifacts.taskrun.format": "<FORMATTER_NAME>"}}'
+```
+
+To select Pipeline-level provenance version
+
+```shell
+kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"artifacts.pipelinerun.format": "<FORMATTER_NAME>"}}'
+```
+
+> Note:\
+> While Chains is able to generate both task-level and pipeline-level provenance at the same time, it's not recommended to enable both because it would be a confusing state that one artifact is linked to 2 provenances.\
```

Comment:
> It's been on my mind to add a `artifacts.tekton.format ` option to configure both of these at once.

This makes sense to me. But this section meant to say it's not recommend to upload both task-level provenance and pipeline-level provenance to a storage backend at the same time. Otherwise, one artifact is linked to 2 different provenances, which is confusing. 

I've clarified a bit. Please let me know if you have other questions.